server/src/modules/mentorship-request/mentorship-request.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import { MentorshipRequestCreateDTO } from './mentorship-request';
+import { MentorshipRequest } from './mentorship-request.model';
+import { InjectModel } from '@nestjs/sequelize';
+import { User } from '../user/user.model';
+const Sequelize = require('sequelize');
+
+@Injectable()
+export class MentorshipRequestService {
+  constructor() {}
+
+  async createMentorshipRequest(
+    mentorshipRequest: MentorshipRequest,
+  ): Promise<MentorshipRequestCreateDTO> {
```

Comment:
The `MentorshipRequestCreateDTO` is supposed to represent the details sent from the client to the server to create a mentorship request: this method should be returning the `MentorshipRequestGetDTO`. In fact though, I would just return a regular `MentorshipRequest` because that's probably better and they're the same. (Do you even need to return anything here though? I think the client just needs to know if the request succeeded, so `Promise<void>` seems fine.)

(In general, I have mixed feelings on whether services should "know" about DTOs. In theory I feel like DTOs are about server-client communication, so only the controllers should know about them and deal with them. However, in practice it could be simpler for services to use them and I'm not an expert here.)